### PR TITLE
:sparkles: feat: add spellcheck ci

### LINKED ISSUE #8: 🛠 Tooling: Investigate adding spell checking to CI
There are a few good libraries floating around. I should a relatively popular one and try it out.

This might be a no-go, if it requires adding a ton of words to an allowlist.

### PR DESCRIPTION
**Pull request by fuadop**
## PR Checklist

- [x] Addresses an existing open issue: fixes #8 
- [x] That issue was marked as [accepting prs](https://github.com/LearningTypeScript/projects/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/LearningTypeScript/projects/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

 uses cspell for spellcheck on projects directory.

### CODE REVIEW

File: .github/workflows/spellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "16"
+          cache: "npm"
+      - run: npm ci
+      - name: Run spellcheck
+        run: npx cspell lint projects/**/*.{ts,md,js} --show-suggestions
```
Comment (JoshuaKGoldberg) [inline .github/workflows/spellcheck.yml:20]: <details>
<summary>There are a ton of false positives getting logged on code files. Examples inside.</summary>

```plaintext
 13/221 ./projects/arrays/text-processor/src/index.test.ts 458.14ms X
/home/runner/work/projects/projects/projects/arrays/text-processor/src/index.test.ts:66:17 - Unknown word (abcdefghi) Suggestions: []
/home/runner/work/projects/projects/projects/arrays/text-processor/src/index.test.ts:66:61 - Unknown word (abcdefghi) Suggestions: []
```

```plaintext
 33/221 ./projects/configuration-options/secret-messages/01-encoded-outputs/index.verify.ts 316.61ms X
/home/runner/work/projects/projects/projects/configuration-options/secret-messages/01-encoded-outputs/index.verify.ts:12:14 - Unknown word (mkstx) Suggestions: [mktx, MKTX, masts, misti, mists]
/home/runner/work/projects/projects/projects/configuration-options/secret-messages/01-encoded-outputs/index.verify.ts:13:15 - Unknown word (uytm) Suggestions: [yum, ubyte, ulta, unto, upto]
```

```plaintext
/home/runner/work/projects/projects/projects/type-operations/template-literal-type-shenanigans/03-split-on/index.test.ts:7:46 - Unknown word (bbaabb) Suggestions: [babas, baba, baobab, blabby, baa's]
/home/runner/work/projects/projects/projects/type-operations/template-literal-type-shenanigans/03-split-on/index.test.ts:8:42 - Unknown word (bbaabb) Suggestions: [babas, baba, baobab, blabby, baa's]
/home/runner/work/projects/projects/projects/type-operations/template-literal-type-shenanigans/03-split-on/index.test.ts:9:48 - Unknown word (ababa) Suggestions: [abba, abaca, aqaba, Abba, Aqaba]
```
</details>

Do you see a lot of benefit in running on `.ts` and `.js` files?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Thanks for sending this in! There are a lot of [failures in CI](https://github.com/LearningTypeScript/projects/runs/6949577086?check_suite_focus=true) right now so we'll have to avoid/fix them in some way.

**Comment by JoshuaKGoldberg** [comment]
👋 @fuadop just checking in, is this still something you have time to look into?

No worries if not, just making sure the issue isn't blocked! 😄

**Comment by fuadop** [comment]
Hey @JoshuaKGoldberg 👋 , super sorry about the delay.

About this, the words truly aren't existent, Do we create a dictionary file? but that's what we are trying to avoid here right?

**Comment by JoshuaKGoldberg** [comment]
Hmm yeah, it is unfortunate how many words are being flagged right now. How about including only `.md` files in the spellcheck check? There are only two `.md` failures right now, both of which should go away once the [`typescript` dictionary](https://cspell.org/docs/dictionaries/#programming-language-dictionaries) is [configured to be used](https://cspell.org/docs/getting-started/#1-create-a-configuration-file).

**Comment by fuadop** [comment]
Hey @JoshuaKGoldberg,
I am actually seeing more than two words being flagged for `.md` files on my end 🤔.

image

**Comment by JoshuaKGoldberg** [comment]
Hmm, is that with the TypeScript dictionary enabled? If so: I probably did it wrong on my end then 😄.

**Comment by fuadop** [comment]
With the typescript dictionary, it reduced to 37 from 42. 
image

**Comment by JoshuaKGoldberg** [comment]
Heh. Well, there are a lot of custom words in this repo: Typarium, Typespeare, etc. I think it makes sense that we'd have to add a dictionary.

If that's a lot more work than you have time for, I can always take over - no worries!

**Comment by fuadop** [comment]
Yea, I can add that. how would you love the dictionary file to be stored in the repo? is the dictionary file okay in the root directory?

**Comment by JoshuaKGoldberg** [comment]
Yeah that makes sense! If you have opinions on a better way I'd be interested though 😄.

File: projects/type-modifiers/modifiers-of-the-types/02-artifact-assertions/characters.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export const characters = [
     }`,
 	`{
         "name": "Teela Na",
-        "powers": ["telekenesis", "telepathy", "teleportation"],
+        "powers": ["telekinesis", "telepathy", "teleportation"],
```
Comment (JoshuaKGoldberg) [inline projects/type-modifiers/modifiers-of-the-types/02-artifact-assertions/characters.ts:24]: Nice 😄 spellcheck really is useful!

**Review by JoshuaKGoldberg — APPROVED** [review]
Wonderful, thanks for this @fuadop! ✨

**Pull request merged**